### PR TITLE
Fix thttpd_alloc.h:64:32: error: 'free' called on pointer 'hc' with nonzero offset 286 [-Werror=free-nonheap-object]

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
Report here: https://github.com/apache/incubator-nuttx/actions/runs/3302940198/jobs/5450348182

## Impact
Minor

## Testing
Pass CI

### CODE REVIEW

File: netutils/thttpd/libhttpd.h [hunk 1]
```diff
@@ -204,9 +211,18 @@ typedef struct
...
+  size_t maxexpnfilename;
+  size_t maxencodings;
+  size_t maxpathinfo;
+  size_t maxquery;
+  size_t maxaccept;
+  size_t maxaccepte;
+  size_t maxreqhost;
+  size_t maxhostdir;
+  size_t maxremoteuser;
+  size_t maxresponse;
```
Comment (pkarashchenko) [inline netutils/thttpd/libhttpd.h:225]: Please fix at line 230 as well

Comment (xiaoxiang781216) [reply netutils/thttpd/libhttpd.h:225]: Done.

**Review by pkarashchenko — APPROVED** [review]

**Comment by pkarashchenko** [comment]
```
./cxxtest_main.cxx:40:10: fatal error: array: No such file or directory
   40 | #include <array>
```

**Comment by xiaoxiang781216** [comment]
let's ignore the download fail and merge this patch to unblock ci:
```
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  264k    0  264k    0     0   572k      0 --:--:-- --:--:-- --:--:--  572k
./cxxtest_main.cxx:40:10: fatal error: array: No such file or directory
   40 | #include <array>
      |          ^~~~~~~
```

**Pull request merged**